src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -27,25 +31,65 @@ namespace CoreNodeModels.Input
     [OutPortDescriptions("Selected Color.")]
     public class ColorPalette : NodeModel
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private DSColor prevColor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
+        public DSColor DsColor
         {
             get { return dscolor; }
             set
-            {                              
+            {
+                if (dscolor.Equals(value) ||prevColor.Equals(value))
```

Comment:
we need to compare with previous color because if the value is same as the previous value or the current value we're potentially in an undo or a redo.

imagine undoing and redoing back and forth